### PR TITLE
[Extensions] Migrated Search Top Anomaly Result to extension

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
This PR migrates Search Top Anomalies Results to extension for the below requests.

- [x] GET  _extensions/_ad-extension/detectors/lxkMxYcBs8GyCDhBVHXy/results/_topAnomalies?historical=true
- [x] GET _extensions/_ad-extension/detectors/lxkMxYcBs8GyCDhBVHXy/results/_topAnomalies?historical=false

### Issues Resolved
https://github.com/opensearch-project/opensearch-sdk-java/issues/379

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshpalis — APPROVED** [review]
LGTM

File: src/main/java/org/opensearch/ad/model/AnomalyResultBucket.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public static AnomalyResultBucket createAnomalyResultBucket(Bucket bucket) {
         return new AnomalyResultBucket(
             bucket.getKey(),
             (int) bucket.getDocCount(),
-            ((InternalMax) bucket.getAggregations().get(MAX_ANOMALY_GRADE_FIELD)).getValue()
+            ((ParsedMax) bucket.getAggregations().get(MAX_ANOMALY_GRADE_FIELD)).getValue()
         );
```
Comment (vibrantvarun) [inline src/main/java/org/opensearch/ad/model/AnomalyResultBucket.java:59]: This changed needs to be document right?

@joshpalis @owaiskazi19

File: src/main/java/org/opensearch/ad/rest/RestSearchTopAnomalyResultAction.java [hunk 2]
```diff
@@ -16,46 +16,68 @@
...
+            AnomalyDetectorExtension.AD_BASE_DETECTORS_URI,
             RestHandlerUtils.DETECTOR_ID,
             RestHandlerUtils.RESULTS,
             RestHandlerUtils.TOP_ANOMALIES
         );
     private final String SEARCH_TOP_ANOMALY_DETECTOR_ACTION = "search_top_anomaly_result";
 
-    public RestSearchTopAnomalyResultAction() {}
+    public RestSearchTopAnomalyResultAction(ExtensionsRunner extensionsRunner, SDKRestClient client) {
+        this.client = client;
```
Comment (vibrantvarun) [inline src/main/java/org/opensearch/ad/rest/RestSearchTopAnomalyResultAction.java:63]: client -> sdkRestClient

File: src/main/java/org/opensearch/ad/transport/SearchTopAnomalyResultTransportAction.java [hunk 3]
```diff
@@ -184,7 +183,7 @@ public class SearchTopAnomalyResultTransportAction extends
     private static final String BUCKET_SORT_FIELD = "bucket_sort";
     public static final String MULTI_BUCKETS_FIELD = "multi_buckets";
     private static final Logger logger = LogManager.getLogger(SearchTopAnomalyResultTransportAction.class);
-    private final Client client;
+    private final SDKRestClient client;
```
Comment (vibrantvarun) [inline src/main/java/org/opensearch/ad/transport/SearchTopAnomalyResultTransportAction.java:186]: client -> sdkRestClient

**Review by vibrantvarun — COMMENTED** [review]
Added some comments

File: src/main/java/org/opensearch/ad/model/AnomalyResultBucket.java (hunk 1 above)
Comment (owaiskazi19) [reply src/main/java/org/opensearch/ad/model/AnomalyResultBucket.java:59]: @vibrantvarun good call. Here you go: https://github.com/opensearch-project/opensearch-sdk-java/pull/717

**Review by dbwiddis — APPROVED** [review]
LGTM!

**Comment by owaiskazi19** [comment]
@vibrantvarun addressed your comments here: https://github.com/opensearch-project/anomaly-detection/pull/885

**Pull request merged**